grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -625,20 +679,12 @@ class StockMovementService {
 
     EditPage getEditPage(String id) {
         EditPage editPage = new EditPage()
-        def startTime = System.currentTimeMillis()
-        StockMovement stockMovement = getStockMovement(id)
-        log.info("Get stock movement ${id}: ${System.currentTimeMillis() - startTime}")
-
-        Map monthlyStocklistQuantities = calculateMonthlyStockListQuantity(stockMovement.origin)
 
-        startTime = System.currentTimeMillis()
-        stockMovement.lineItems.each { stockMovementItem ->
-            stockMovementItem.stockMovement = stockMovement
-            EditPageItem editPageItem = buildEditPageItem(stockMovementItem)
-            editPageItem.quantityConsumed = monthlyStocklistQuantities.get(stockMovementItem.product.id)
-            editPage.editPageItems.addAll(editPageItem)
+        if (!grailsApplication.config.openboxes.api.pagination.enabled) {
```

Comment:
I've reviewed this line a few times since last week and I'm having trouble understanding the point of this condition. And more to the point, what happens when pagination is actually enabled. Seems the API would return an empty response.

I was expecting the pagination to happen while retrieving stock movement items within the getStockMovement() method. And no matter whether all stock movement items were returned or not, we'd apply the same logic to build the edit page items or pick page items.

Reply:
And actually I think when I asked for a configuration property I was thinking that the client would pull the config property (from whatever endpoint exposes config properties - getAppContext()) and if the config property was enabled then the client would pass the max and offset. If the property was disabled it wouldn't. The endpoint itself would delegate to the service and the service would determine whether to paginate or not based on the presence of max and offset parameters.

Reply:
https://discuss.jsonapi.org/t/what-is-a-good-pagination-strategy-for-nested-collections/1034/3

Reply:
The response in that post reframes the problem. Maybe we shouldn't be paginating on a nested resource, but rather on the main resource using a filter for the parent stock movement ID. That way we don't have to deal with two different pagination mechanisms for the same requests. 
```
/api/stockMovementItem?parentId=:parentId
```

Reply:
@DominikaHatala @awalkowiak @pmuchowski Thoughts?

Reply:
The point of this condition is to not include line items for getStockMovement() if pagination is enabled. We have sepeare ednpoints then - one for stock movement header data (getStockMovement) and the other is endpoint for getting paginated line items (with max and offset parameter). If pagination is disabled then those requests for paginated items are not send and stock movement items are included in getStockMovement() method

Reply:
I agree with @jmiranda with this one. It is up to endpoint to decide if he wants to have paginated resources. If you look into PagedResultList for criteria builder here: https://grails.github.io/grails2-doc/1.3.9/ref/Domain%20Classes/createCriteria.html, if you pass the necessary pagination values as arguments, then it will be paginated, otherwise not. In other case if pagination is enabled and user want a response with all rows, he'd need to make one request to check what totalCount is and then make another request to get all rows.
In your case @DominikaHatala, we need to check if response for paginated items is empty and then go back and extract items from getStockMovement response which was done earlier.
@jmiranda I think that having two separate requests is fine (one for parent info and second for items), but each needs to return something.
@DominikaHatala Checking if line items are empty in second request then go back to first and extract line items is weird workflow and I don't like it. There should be simplest and the most stright forward approach when parsing responses: getStockMovement -> getStockMovementItems.

Reply:
@awalkowiak Yes agreed. Is it possible to never return stock movement items when getting the header. In other words, we should ALWAYS have a second request to get stock movement items (whether paged or not).
@DominikaHatala Is that possible? Are there challenges within the API client if we do that?